### PR TITLE
Upgrade Python Modeule to PostgreSQL 9.2

### PR DESCRIPTION
**Pull request by HaozhouWang**
1. Backport GPDB related codes
2. Backport multiple array support
3. Backport cancel support and 64-bit SPI
4. Fix errors due to files spilt in PG 9.2
5. Fix wrong answer files and follow upsteam answers

### CODE REVIEW

File: src/pl/plpython/plpy_spi.c [hunk 1]
```diff
@@ -54,11 +55,15 @@ PLy_spi_prepare(PyObject *self, PyObject *args)
 	{
 		PLy_exception_set(PyExc_TypeError,
 					   "second argument of plpy.prepare must be a sequence");
+		PLy_enter_python_intepreter = true;
```
Comment (zhangh43) [inline src/pl/plpython/plpy_spi.c:60]: lack PLy_enter_python_intepreter= false at the beginning of spi prepare

File: src/pl/plpython/plpy_subxactobject.c [hunk 2]
```diff
@@ -93,6 +94,7 @@ PLy_subtransaction_new(PyObject *self, PyObject *unused)
 	ob->started = false;
 	ob->exited = false;
 
+	PLy_enter_python_intepreter = true;
```
Comment (zhangh43) [inline src/pl/plpython/plpy_subxactobject.c:97]: did you process PLy_enter_python_intepreter in PLy_output?

**Comment by Eulerizeit** [comment]
My understanding was that the PL/Python version we were using was close to Postgres 10. I'm under the impression that the PL/* code is fairly well separated from the core Postgres code so getting up to the latest would be ideal. 

@danielgustafsson am I thinking about this correctly?

**Comment by HaozhouWang** [comment]
@Eulerizeit I had a look PG 10 and PG9.2 codes, and found that PL/* code used a lot of GPDB/Postgres APIs, and such interface will changed/updated in the following versions, and many of PL/* commits are to update to suit these interface changes. I think PL/* code can follow the GPDB merge to minimum the conflict. 
@paul-guo- @guofengrichard @baobao0206

File: src/pl/plpython/plpy_spi.c (hunk 1 above)
Comment (HaozhouWang) [reply src/pl/plpython/plpy_spi.c:60]: Thanks, fixed

File: src/pl/plpython/plpy_subxactobject.c (hunk 2 above)
Comment (HaozhouWang) [reply src/pl/plpython/plpy_subxactobject.c:97]: Thanks, fixed

**Pull request merged**